src/compiler/checker.ts

```diff
@@ -336,6 +341,13 @@ namespace ts {
                         target[id] = source[id];
                     }
                     else {
+                        if (target === globals && source !== builtinGlobals) {

```

Comment:
so why not just let them merge, then later on make sure that the undefined symbol is the one we have created, or that it does not have any declarations?